components/EndlessDocumentList.vue

```diff
@@ -0,0 +1,65 @@
+<script setup lang="ts">
+import type { IAllDocumentModels } from "@bcc-code/bmm-sdk-fetch";
+import { useInfiniteScroll } from "@vueuse/core";
+
+const props = defineProps<{
+  load: (skip: number, take: number) => Promise<IAllDocumentModels[]>;
+}>();
+
+const list = ref<IAllDocumentModels[]>([]);
+const loadingMore = ref(false);
+let position = 0;
+const fullyLoaded = ref(false);
+onMounted(() => {
+  const main = ref<HTMLElement | null>(document.querySelector("main"));
+  useInfiniteScroll(
+    main,
+    () => {
+      if (loadingMore.value || fullyLoaded.value) {
+        return;
+      }
+
+      loadingMore.value = true;
+      props
+        .load(position, 40)
+        .then((data) => {
+          position += 40;
+          if (data) {
+            if (data.length === 0) {
+              console.log(`list is fully loaded. position: ${position}`);
+              fullyLoaded.value = true;
+            }
+
+            list.value = list.value.concat(data);
+          }
+          loadingMore.value = false;
+        })
+        .catch((e) => {
+          console.error("error", e);
+        });
+    },
+    { distance: 10, interval: 500, canLoadMore: () => !fullyLoaded.value },
+  );
+});
+watch(
+  [useNuxtApp().$i18n.locale, () => contentLanguageStore().contentLanguages],
+  async () => {
+    const data = await props.load(0, position);
```

Comment:
We should also handle errors here... You do it well above, but here you leave errors to the watch-implementation to deal with.

Reply:
we're just swallowing the exception.
If it happens often, we should invest some time to figure out what's the best user experience.